src/test/java/com/cag/cagbackendapi/integration/UserControllerIntegrationTests.kt

```diff
@@ -139,6 +141,103 @@ class UserControllerIntegrationTests {
+    }
+
+    @Test
+    fun updateUser_missingUserId_400BadRequest() {
+        val invalidUpdateUser = UserResponseDto(null, first_name = "Tony", last_name = "Stark", email="tstark@gmail.com")
+        val headers2 = HttpHeaders()
+        headers2.set("authKey", validAuthKey)
+        val request2 = HttpEntity(invalidUpdateUser, headers2)
+
+        val errorDetailsResponse = testRestTemplate.exchange("/user/", HttpMethod.PUT, request2, ErrorDetails::class.java)
+
+        assertEquals(HttpStatus.BAD_REQUEST, errorDetailsResponse.statusCode)
+        assertNotNull(errorDetailsResponse?.body?.time)
+        assertEquals(errorDetailsResponse?.body?.restErrorMessage, RestErrorMessages.BAD_REQUEST_MESSAGE)
+        assertEquals(errorDetailsResponse?.body?.detailedMessage, DetailedErrorMessages.INVALID_UUID)
+    }
+
+    @Test
+    fun updateUser_invalidAuthKey_401Unauthorized() {
+        val headers = HttpHeaders()
+        headers.set("authKey", validAuthKey)
+        val request = HttpEntity(validTestUser, headers)
+
+        val createdUserResponse = testRestTemplate.postForEntity("/user/register", request, String::class.java)
+        val createUser = objectMapper.readValue(createdUserResponse.body, UserResponseDto::class.java)
+        val userId = createUser.user_id
+
+        val validUpdateUser = UserResponseDto(user_id = userId, first_name = "Tony", last_name = "Stark", email="tstark@gmail.com")
+        val headers2 = HttpHeaders()
+        headers2.set("authKey", "invalidAuthKey")
+        val request2 = HttpEntity(validUpdateUser, headers2)
+
+        val errorDetailsResponse = testRestTemplate.exchange("/user/", HttpMethod.PUT, request2, ErrorDetails::class.java)
+
+        assertEquals(HttpStatus.UNAUTHORIZED, errorDetailsResponse.statusCode)
+        assertNotNull(errorDetailsResponse?.body?.time)
+        assertEquals(errorDetailsResponse?.body?.restErrorMessage, RestErrorMessages.UNAUTHORIZED_MESSAGE)
+        assertEquals(errorDetailsResponse?.body?.detailedMessage, DetailedErrorMessages.WRONG_AUTH_KEY)
+    }
+
+    //getting a 200 error
+    @Test
+    fun updateUser_userNotFound_404NotFound(){
+        val validUpdateUser = UserResponseDto(user_id = UUID.randomUUID(), first_name = "Tony", last_name = "Stark", email="tstark@gmail.com")
+        val headers2 = HttpHeaders()
+        headers2.set("authKey", validAuthKey)
+        val request2 = HttpEntity(validUpdateUser, headers2)
+
+        val errorDetailsResponse = testRestTemplate.exchange("/user/", HttpMethod.PUT, request2, ErrorDetails::class.java)
+
+        assertEquals(HttpStatus.BAD_REQUEST, errorDetailsResponse.statusCode)
+        assertNotNull(errorDetailsResponse?.body?.time)
+        assertEquals(errorDetailsResponse?.body?.restErrorMessage, RestErrorMessages.NOT_FOUND_MESSAGE)
+        assertEquals(errorDetailsResponse?.body?.detailedMessage, DetailedErrorMessages.USER_NOT_FOUND)
+
+    }
+
+    @Test
+    fun updateUser_internalServerError_500Error(){}
```

Comment:
You can delete this because this case should not be possible to integration test for since you shouldn't be able to send data that causes this issue. We can unit test for this since we use mocking! But not integration test. The same is true for the service unavailable test.